canvas_modules/common-canvas/src/common-properties/components/moveable-table-rows/moveable-table-rows.jsx

```diff
@@ -166,17 +213,29 @@ class MoveableTableRows extends React.Component {
 		const selected = this.props.controller.getSelectedRows(this.props.propertyId).sort(function(a, b) {
 			return a - b;
 		});
+		const leastValue = this.getLeastValue();
+		const staticRows = this.props.controller.getStaticRows(this.props.propertyId).sort();
 		// only move up if not already at the top especially for multiple selected
-		if (selected.length !== 0 && selected[0] !== 0) {
+		// Move up only till the static rows index
+		if (selected.length !== 0 && selected[0] > leastValue) {
 			const controlValue = this.props.controller.getPropertyValue(this.props.propertyId);
 			for (var i = 0; i <= selected.length - 1; i++) {
-				const selectedRow = selected.shift();
-				if (selectedRow !== 0) {
+				if (staticRows.length > 0 && selected[0] > leastValue + 1) {
+					const selectedRow = selected.shift();
 					const tmpRow = controlValue[selectedRow - 1];
 					controlValue[selectedRow - 1] = controlValue[selectedRow];
 					controlValue[selectedRow] = tmpRow;
 					selected.push(selectedRow - 1);
 					this.props.controller.moveErrorMessageRows(this.props.propertyId.name, selectedRow, selectedRow - 1);
+				} else if (staticRows.length === 0) {
+					const selectedRow = selected.shift();
```

Comment:
Are you able to combine these if statements as well? Is there a reason why this line `const selectedRow = selected.shift();` is outside the if statement?

Reply:
We need this condition for row move up. If I remove this the behavior is incorrect.